bambi/models.py

```diff
@@ -746,9 +746,9 @@ def __init__(self, name, data, categorical=False, random=False, prior=None,
 
         # For the sake of computational efficiency (i.e., to avoid lots of
         # large matrix multiplications in the backends), invert the dummy-
-        # coding process and represent binary categoricals as a vector of
+        # coding process and represent full-rank dummies as a vector of
         # indices into the coefficients.
-        if data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
+        if constant and data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
```

Comment:
Thinking about this also gave me the idea that we might be able to efficiently handle big random terms generally (i.e., not just random intercepts but also random slopes) by using some basic sparse matrix multiplication techniques, which is essentially what we are doing now. It looks like scipy has some sparse matrix classes, but we might not need all that machinery if the only two things we need to do are represent the data and multiply it by a vector. I found [this nice page](https://www.cs.cmu.edu/~scandal/cacm/node9.html) which makes me think it might actually be fairly easy to do ourselves for cheap. Could be worth exploring more.